### PR TITLE
Pop up v2 and clean up

### PR DESCRIPTION
**Pull request by apple00juice**
I wanted to create a popUp with a backdrop and make it more general, thus I created a new one based on luis's initial design. Just some example popUps:
NameErrorScreenshot 2022-04-10 160052updating

I updated the popUp in CreateLobby, UpdateLobby, PublicLobbies, Lobby. Additionally I looked at some errors that were thrown in the console and code smells of sonarCloud. And updated the homepage to use the header and added a cool animation 😃

### CODE REVIEW

**Review by paolovic — COMMENTED** [review]
great!
Could you split it up into smaller pieces next time, please? 😅

**Comment by luis-tm** [comment]
I modified the png file hannibal_header to make the background transparent.

**Comment by apple00juice** [comment]
> great! Could you split it up into smaller pieces next time, please? 😅

yea, its a bit big. I tried to split it a bit with the commits ^^

File: src/components/ui/CustomPopUp.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+CustomPopUp.propTypes = {
+    open: PropTypes.bool,
+    information: PropTypes.any.isRequired,
+    children: PropTypes.node,
+    onClick: PropTypes.func,
+    onMouseDown: PropTypes.func,
+    onMouseUp: PropTypes.func,
+}
+
+export default CustomPopUp;
```
[the remark this answers is not in the document]
Comment (apple00juice) [reply src/components/ui/CustomPopUp.js]: Added a line

**Review by luis-tm — APPROVED** [review]

**Review by mkorob — APPROVED** [review]
Animation is a nice idea!

**Comment by apple00juice** [comment]
thx for reviewing it 😃

**Pull request merged**